src/pages/home/report/ParticipantLocalTime.js

```diff
@@ -35,13 +35,18 @@ class ParticipantLocalTime extends React.Component {
         }, 1000));
     }
 
+    shouldComponentUpdate(nextProps, nextState) {
+        return nextState.localTime !== this.state.localTime;
+    }
+
     componentWillUnmount() {
         clearInterval(this.timer);
         clearInterval(this.readyTimer);
     }
 
     getParticipantLocalTime() {
         const reportRecipientTimezone = lodashGet(this.props.participant, 'timezone', {});
+        moment.locale(this.props.preferredLocale);
```

Comment:
This will cause issues with moment() usages. it updates the locale to another user's timezone globally. There are many places where we are dependent on the moment() to get the time.

Do you think it's safe to use it this way?

Reply:
@parasharrajat I haven't reviewed this code yet, in the middle of testing. Can you give me an example of a type of regression this could cause? So far in my testing things are working great and haven't noticed issues with displayed times yet.

Reply:
Please look at ReimbursementAccountPage L174 & CompanyStep L107. Dates could be +1 or -1 based on timezone etc.